src/gravity/gravity.hpp

```diff
@@ -46,11 +50,19 @@ class Gravity {
   void Initialize(ParameterInput *pin);
   void Solver(const AthenaArray<Real> &u);
 
+  void EnrollSource(AthenaArray<Real> &arr, int idx);
+  void UnenrollSource(AthenaArray<Real> &arr, int idx);
+  void ComputeSource();
+
   friend class MGGravityDriver;
 
  private:
   bool gravity_tensor_momentum_;
   bool gravity_tensor_energy_;
+
+  // actual copy of accumulated source and (pointers, index) pair to enrolled sources
+  AthenaArray<Real> accumulated_src_;
+  std::vector<std::pair<AthenaArray<Real>*, int>> enrolled_src_;
```

Comment:
I see this way of pointing the source array confusing. Isn't it better to point actual source array (e.g., density) rather than pointing 4D array with index?